charts/clusterapi-resources/templates/KubeadmConfigTemplate.yaml

```
@@ -15,6 +15,8 @@ spec:
           kubeletExtraArgs:
             cloud-provider: external
           name: '{{`{{ local_hostname }}`}}'
+          taints:
+          {{- toYaml ( $md.joinConfiguration.nodeRegistration.taints | default list ) | nindent 10 }}
```

Should we do this or should we merge the two `joinConfiguration` structures to make this more generic?

---

I wanted to gather yalls opinions/thoughts on the changes. I have no preference with a more generic approach, as long as @iblackman doesn't mind. Just wanted to start with the bare minimum

---

My preference is to be specific because that way, you leave almost no room for errors on whoever is passing values

---

@iblackman Do you prefer the way it is now vs a top-level $md.taints key?

---

But I'm okay with the merging options too. It is just that by default, helm values in general follow the override strategy.

---

```suggestion
          {{- toYaml ( $md.taints | default list ) | nindent 10 }}
```

This or what Ryan wrote @iblackman ? Let's ship this.